### PR TITLE
Implemented signals to detect when an investor follows a project or s…

### LINKED ISSUE #62: Implement Notification Trigger Mechanism [21-2 Notification Trigger Mechanisms & Services]
**Description:**
Develop backend mechanisms that trigger notifications based on specific actions by an investor, such as following a startup's profile or sending a message.

**Acceptance Criteria:**
- Implement event listeners or hooks within the application that detect when an investor follows a profile or sends a message.
- Upon detecting such events, a notification record should be created in the database.
- Ensure that the notification mechanism is efficient and does not introduce significant latency to the investor's actions.

Task #21

### PR DESCRIPTION
**Pull request by avu-ua**
…ubscribes for a share in it.

### CODE REVIEW

File: ForumProject/notifications/signals.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Returns:
+    - Notification: The created `Notification` instance.
+    '''
+    Notification.objects.create(
+            project=instance.project,
+            startup=instance.project.startup,
+            investor=instance.investor,
+            trigger='follower(s) list changed' if follow_status_change else 'subscription changed',
+            initiator='investor',
+        )
```
Comment (mehalyna) [inline ForumProject/notifications/signals.py]: Consider adding validation to ensure all fields (`project`, `startup`, `investor`, etc.) are present and valid

File: ForumProject/notifications/signals.py
Comment (mehalyna) [inline ForumProject/notifications/signals.py:1]: 1) If additional metadata (like `timestamp`) is required, pass relevant parameters directly to the notification creation function.
2) Consider implementing bulk notification creation if multiple changes are processed simultaneously.

**Review by mehalyna — CHANGES_REQUESTED** [review]

File: ForumProject/notifications/signals.py (hunk 1 above)
Comment (avu-ua) [reply ForumProject/notifications/signals.py]: Done

File: ForumProject/notifications/signals.py
Comment (avu-ua) [reply ForumProject/notifications/signals.py:1]: On point 1 - class Notification does have a `models.DateTimeField`, however it has parameter `auto_now_add=True`, therefore I do not think passing any parameters to the notification creation function is required.
Point 2 - done.

**Review by mehalyna — APPROVED** [review]

**Pull request merged**